### PR TITLE
E2E - (DESTINATARIO) Dettaglio notifica

### PR DESCRIPTION
**Pull request by azhansultan**
* Added Web elements
* Added new step definitions 
*  Added new feature file

### CODE REVIEW

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/AccediAPiattaformaNotifichePage.java [hunk 1]
```diff
@@ -19,6 +16,71 @@ public class AccediAPiattaformaNotifichePage extends BasePage {
...
+    @FindBy(css = "[data-testid='more-less-timeline-step']")
+    WebElement vediPiuDettagli;
+
+    @FindBy(linkText = "Vedi meno dettagli")
+    WebElement vediMenoDettagliButton;
+
+    @FindBy(css="[type='application/pdf']")
+    WebElement pdfType;
+
+    @FindBy(css = "[data-testid='documentButton']")
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/AccediAPiattaformaNotifichePage.java]: non dimenticare id aggiornare il documento

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/AccediAPiattaformaNotifichePage.java [hunk 2]
```diff
@@ -40,4 +102,143 @@ public void waitLoadAccediAPiattaformaNotifichePage(){
...
+        getWebDriverWait(30).withMessage("Il bottone recapiti non è visibile").until(ExpectedConditions.visibilityOf(recapitiButton));
+        this.js().executeScript("arguments[0].click()", this.recapitiButton);
+    }
+
+    public void clickNotificheButton() {
+        getWebDriverWait(30).withMessage("Il bottone notifiche non è visibile").until(ExpectedConditions.visibilityOf(notificheButton));
+        notificheButton.click();
+    }
+
+    public void clickatestazionePersalvare() {
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/AccediAPiattaformaNotifichePage.java]: fix in "clickAttestazionePerSalvare"

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/AccediAPiattaformaNotifichePage.java [hunk 3]
```diff
@@ -40,4 +102,143 @@ public void waitLoadAccediAPiattaformaNotifichePage(){
...
+        getWebDriverWait(30).withMessage("Il attestazione non è visibile").until(ExpectedConditions.visibilityOf(attestazione));
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/AccediAPiattaformaNotifichePage.java]: fix in "l'attestazione non é visiibile"

File: src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/AccediAPiattaformaNotifichePage.java [hunk 4]
```diff
@@ -40,4 +102,143 @@ public void waitLoadAccediAPiattaformaNotifichePage(){
...
+        return getWebDriverWait(30).withMessage("Il sezione scarica avviso non è visibile").until(ExpectedConditions.visibilityOf(scaricaAvviso)).isDisplayed();
+    }
+
+    public boolean pagaAvvisoDisplayed() {
+        return getWebDriverWait(30).withMessage("Il sezione paga avviso non è visibile").until(ExpectedConditions.visibilityOf(pagaAvviso)).isDisplayed();
+    }
+
+    public boolean sezioneRecapitiDisplayed() {
+        return getWebDriverWait(30).withMessage("Il sezione recapiti non è visibile").until(ExpectedConditions.visibilityOf(sezioneRecapiti)).isDisplayed();
+    }
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/AccediAPiattaformaNotifichePage.java]: inserire un /n tra i metodi

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/LoginPersonaFisicaPagoPA.java [hunk 5]
```diff
@@ -8,6 +8,7 @@
 import it.pn.frontend.e2e.api.personaFisica.SpidDemoLogin;
 import it.pn.frontend.e2e.api.personaFisica.SpidDemoStart;
 import it.pn.frontend.e2e.api.personaFisica.SpidLogin;
+import it.pn.frontend.e2e.common.NotificheDestinatarioPage;
```
Comment (leleOFA) [inline src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/LoginPersonaFisicaPagoPA.java]: attenzione a questo import se non usato cancellare

Comment (azhansultan) [reply src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/LoginPersonaFisicaPagoPA.java]: okay, grazie mille per feedback

**Comment by afus0** [comment]
@azhansultan, the `develop` branch has been updated. Please make sure to resolve any conflicts you encounter during the update.
Check if the test works properly once you updated.

P.S: Login steps have changed:

- PA now doesn't require any additional file specification
- PF & PG must have "delagato" or "delegante" in order to take the respective `tokenExchange`
E.g.: 
`    Given PF - Si effettua la login tramite token exchange come "delegante", e viene visualizzata la dashboard
`

**Comment by azhansultan** [comment]
@ImJimMoriarty **Thank you for update. I have updated my branch, conflicts are resolved and steps are updated. But when I try to run a test from feature file, I still get the following error because of **driver=null**** 
   io.cucumber.core.exception.CucumberException: Failed to instantiate class it.pn.frontend.e2e.stepDefinitions.destinatario.personaGiuridica.LoginPGPagoPATest
	at io.cucumber.core.backend.DefaultObjectFactory.cacheNewInstance(DefaultObjectFactory.java:67)
	at io.cucumber.core.backend.DefaultObjectFactory.getInstance(DefaultObjectFactory.java:45)
	at io.cucumber.java.AbstractGlueDefinition.invokeMethod(AbstractGlueDefinition.java:47)
	at io.cucumber.java.JavaStepDefinition.execute(JavaStepDefinition.java:29)
	at io.cucumber.core.runner.CoreStepDefinition.execute(CoreStepDefinition.java:66)
	at io.cucumber.core.runner.PickleStepDefinitionMatch.runStep(PickleStepDefinitionMatch.java:63)
	at io.cucumber.core.runner.ExecutionMode$1.execute(ExecutionMode.java:10)
	at io.cucumber.core.runner.TestStep.executeStep(TestStep.java:84)
	at io.cucumber.core.runner.TestStep.run(TestStep.java:56)
	at io.cucumber.core.runner.PickleStepTestStep.run(PickleStepTestStep.java:51)
	at io.cucumber.core.runner.TestCase.run(TestCase.java:84)
	at io.cucumber.core.runner.Runner.runPickle(Runner.java:75)
	at io.cucumber.core.runtime.Runtime.lambda$executePickle$6(Runtime.java:132)
	at io.cucumber.core.runtime.CucumberExecutionContext.lambda$runTestCase$5(CucumberExecutionContext.java:130)
	at io.cucumber.core.runtime.RethrowingThrowableCollector.executeAndThrow(RethrowingThrowableCollector.java:23)
	at io.cucumber.core.runtime.CucumberExecutionContext.runTestCase(CucumberExecutionContext.java:130)
	at io.cucumber.core.runtime.Runtime.lambda$executePickle$7(Runtime.java:132)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.cucumber.core.runtime.Runtime$SameThreadExecutorService.execute(Runtime.java:253)
	at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:123)
	at io.cucumber.core.runtime.Runtime.lambda$runFeatures$3(Runtime.java:114)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.SliceOps$1$1.accept(SliceOps.java:200)
	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1602)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at io.cucumber.core.runtime.Runtime.runFeatures(Runtime.java:115)
	at io.cucumber.core.runtime.Runtime.lambda$run$0(Runtime.java:84)
	at io.cucumber.core.runtime.Runtime.execute(Runtime.java:99)
	at io.cucumber.core.runtime.Runtime.run(Runtime.java:82)
	at io.cucumber.core.cli.Main.run(Main.java:87)
	at io.cucumber.core.cli.Main.main(Main.java:30)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at io.cucumber.core.backend.DefaultObjectFactory.cacheNewInstance(DefaultObjectFactory.java:53)
	... 37 more
Caused by: java.lang.NumberFormatException: Cannot parse null string
	at java.base/java.lang.Integer.parseInt(Integer.java:630)
	at java.base/java.lang.Integer.parseInt(Integer.java:786)
	at it.pn.frontend.e2e.common.BasePage.<init>(BasePage.java:18)
	at it.pn.frontend.e2e.section.destinatario.personaGiuridica.HeaderPGSection.<init>(HeaderPGSection.java:19)
	at it.pn.frontend.e2e.stepDefinitions.destinatario.personaGiuridica.LoginPGPagoPATest.<init>(LoginPGPagoPATest.java:36)
	... 43 more

**Review by leleOFA — APPROVED** [review]

**Pull request merged**